### DISCUSSION TITLE
Model at each component

### DISCUSSION THREAD

**Discussion by sonynavdeep81** [explosion/spaCy#9995] [Help: Model Advice] [answered]
I have a very basic question regarding spaCy pipeline. I just want to know that for various components in the pipeline do we have different model for each component which can be trained separately from others?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @sonynavdeep81 ! 

Each component has its own internal state, and it's possible to train them separately. However, some of these components require a [Tok2Vec](https://spacy.io/api/tok2vec), meaning you can't just train them completely alone. You can see more from the [training pipeline design](https://spacy.io/models#design), and in the [general pipeline docs](https://spacy.io/usage/processing-pipelines).

**Reply by sonynavdeep81** [reply]
Thanks a lot!!
